File: src/Bzn.Cloudios.Domain/Entities/Container.cs
using Bzn.Cloudios.Domain.Enums;

namespace Bzn.Cloudios.Domain.Entities;

public sealed class Container
{
    public Guid Id { get; set; }
    public Guid RealmId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? DockerContainerId { get; set; }
    public string ImageName { get; set; } = string.Empty;
    public int InternalPort { get; set; } = 8080;
    public ContainerStatus Status { get; set; } = ContainerStatus.Stopped;
    public double CpuLimitCores { get; set; } = 0.5;
    public long MemoryLimitBytes { get; set; } = 536870912;
    public decimal CostPerHourBRL { get; set; } = 0.02m;
    public DateTime? StartedAtUtc { get; set; }
    public DateTime CreatedAt { get; set; }

    public Realm Realm { get; set; } = null!;
    public List<ContainerVolume> Volumes { get; set; } = [];
    public List<ContainerEnvVar> EnvironmentVariables { get; set; } = [];
}

File: src/Bzn.Cloudios.WebApp/src/types/container.ts
export interface ContainerListResponse {
  items: ContainerListItem[];
  totalCount: number;
  page: number;
  pageSize: number;
  hasNextPage: boolean;
}

export interface ContainerListItem {
  id: string;
  name: string;
  imageName: string;
  internalPort: number;
  status: string;
  cpuLimitCores: number;
  memoryLimitBytes: number;
  costPerHourBRL: number;
  currentMonthCostBRL: number;
  publicUrl?: string;
  startedAtUtc?: string;
  createdAt: string;
  networkName?: string;
}

export interface ContainerDetailResponse {
  id: string;
  name: string;
  imageName: string;
  internalPort: number;
  status: string;
  cpuLimitCores: number;
  memoryLimitBytes: number;
  costPerHourBRL: number;
  currentMonthCostBRL: number;
  dockerContainerId?: string;
  startedAtUtc?: string;
  createdAt: string;
  volumes: ContainerVolumeDto[];
  environmentVariables: object[];
}

export interface CreateContainerRequest {
  name: string;
  imageName: string;
  internalPort: number;
  hostPort?: number;
  networkName: string;
  cpuLimitCores: number;
  memoryLimitBytes: number;
  costPerHourBRL: number;
  environmentVariables: Record<string, string>;
}

export interface ContainerActionResponse {
  id: string;
  status: string;
  dockerContainerId?: string;
  startedAtUtc?: string;
}

export interface ContainerVolumeDto {
  id: string;
  hostPath: string;
  containerPath: string;
  isReadOnly: boolean;
}

export interface ContainerVolumeRequest {
  hostPath: string;
  containerPath: string;
  isReadOnly: boolean;
}

export interface ContainerEnvVarDto {
  id: string;
  key: string;
  value: string;
}

export interface ContainerEnvVarSecureDto {
  id: string;
  key: string;
  value: string;
}

export interface ContainerLogsResponse {
  containerId: string;
  logs: ContainerLogEntry[];
}

export interface ContainerLogEntry {
  timestamp: string;
  stream: string;
  line: string;
}

export interface AdminContainerListResponse {
  items: AdminContainerListItem[];
  totalCount: number;
  page: number;
  pageSize: number;
  hasNextPage: boolean;
}

export interface AdminContainerListItem {
  id: string;
  realmId: string;
  realmName: string;
  name: string;
  imageName: string;
  status: string;
  cpuLimitCores: number;
  memoryLimitBytes: number;
  costPerHourBRL: number;
  currentMonthCostBRL: number;
}
